repo/README.md

```diff
@@ -83,25 +83,29 @@ Below shortcuts are examples, your milage may vary.
 
 #### Windows:
 
-Cygwin is required, and repo must be called explicitly using Cygwin's `bash`.
+Cygwin or Windows Subsystem for Linux is required, and repo must be called explicitly using Cygwin's or Linux's `bash`.
 
-Make sure the folder containing `repo` is in cygwin's `$PATH` variable. Alternatively replace `-l repo` in the parameters below with the full absolute path, for example `-l c:\repoinstallfolder\repo`.
+Make sure the folder containing `repo` is in Windows' `$Path` variable. Alternatively replace `repoWrapper.ps1` in the parameters below with the full absolute path, for example `C:\tools\repoWrapper.ps1`.
 
 * put
-  * Program: `<path-to-cygwin>\bin\bash` - for example: `C:\cygwin64\bin\bash`
-  * Parameters: `-l repo put -f $FilePath$`
+  * Program: `powershell`
```

Comment:
I think rather than replacing the cygwin instructions here, I would have 2 "Windows - powershell" and "Windows - cygwin" sections. The text change above implies the same.

Do you think the powershell approach is easier/better than cygwin nowadays? What is the minimum Windows version requirement?

Reply:
Actually, I had to resort to a Windows shell wrapper in cases of both Cygwin and Windows Subsystem for Linux (WSL). So, my description applies to both. 

- To use Cygwin instead of WSL, one has to replace "mnt" by "cygdrive" as the second parameter.
- I implemented my scripts in Powershell, hence is "powershell" as the Program value. Could have been "cmd" instead.
- I was unable to use the "repo" script for Cygwin out of the box because of the "repo: not inside a vault checkout with a jcr_root base directory" error. Running "repo" from a Powershell script (with the slashes flipped) fixed the problem.
- Using WSL, which is native to Windows, is preferable to using Cygwin IMO. However, your organization can disable installing this subsystem. So, using Cygwin is still a valid option. Finally, corporate security preventing you from installing Cygwin is not unheard of, either.
- WSL is available on Windows 10 since the Anniversary Update on August 2, 2016.

Reply:
I am not sure everyone uses Powershell, as you say. Hence both options should be described.

The particular issue you saw with cygwin (not inside a vault checkout) might need its own fix. Would you mind creating an issue for it?

Reply:
The options are Powershell vs. cmd Windows shells. One can be uncomfortable writing Powershell scripts, but using them should not be a problem. My script uses Powershell functions and cannot be run in cmd. Anybody wishing to use cmd, would have write their own script. Hence, I cannot describe a solution that does not exist.

The particular issue I saw with Cygwin is already described in **$FileDir$ path is giving wrong path in the command** #19. The error message is the same.

Reply:
That problem might be due to quite a few parts of `repo` relying on unix-style paths (`/some/dir`) and not working with windows-style paths (`c:\some\dir`). I don't remember if Cygwin clearly maps everything to unix-style or if someone windows-style paths can come in. Looking at #19 this might be simply arguments set in windows-paths by the IDE (`$FilePath$` in intellij).

Reply:
Moved the code converting input arguments to the UNIX version (if on Cygwin) from the Powershell wrapper script `repoWin.ps1` to `repo` . For WSL, I still have to convert $FilePath$ in  `repoWin.ps1` . FilePath is passed into `repo` with all the backslashes stripped.